src/pharmpy/tools/covsearch/tool.py

```diff
@@ -266,91 +315,49 @@ def filter_search_space_and_model(search_space, model):
     if isinstance(search_space, str):
         search_space = ModelFeatures.create_from_mfl_string(search_space)
     ss_mfl = search_space.expand(filtered_model)  # Expand to remove LET/REF
-    model_mfl = ModelFeatures.create_from_mfl_string(get_model_features(filtered_model))
 
-    # Remove all covariates not part of the search space
-    to_be_removed = model_mfl - ss_mfl
-    covariate_list = to_be_removed.mfl_statement_list(["covariate"])
+    # Clean up all covariate effect in model
+    model_mfl = ModelFeatures.create_from_mfl_string(get_model_features(filtered_model))
+    # covariate effects not in search space, should be kept as it is
+    covariate_to_keep = model_mfl - ss_mfl
+    # covariate effects in both model and search space, should be removed for exploration in future searching steps
+    covariate_to_remove = model_mfl - covariate_to_keep
+    covariate_to_remove = covariate_to_remove.mfl_statement_list(["covariate"])
```

Comment:
Are covariate effects not removed any longer with this? It is intentional to remove covariate effects not in the search space.

Reply:
After discussion we should keep this. Problems with allometric scaling was solved, but instead now covariate effects not in the search space, but in the model will be kept. We should fix that problem later.